config/src/main/java/com/yahoo/config/subscription/impl/ConfigSubscription.java

```diff
@@ -129,50 +159,52 @@ public boolean equals(Object o) {
         return false;
     }
 
-    void setConfigChanged(boolean changed) {
-        this.configChanged = changed;
-    }
-
-    void setGenerationChanged(boolean genChanged) {
-        this.generationChanged = genChanged;
-    }
 
     /**
      * Called from {@link ConfigSubscriber} when the changed status of this config is propagated to the clients
      */
-    public void resetChangedFlags() {
-        setConfigChanged(false);
-        setGenerationChanged(false);
+    public boolean isConfigChangedAndReset() {
+        ConfigState<T> prev = config.get();
+        while ( !config.compareAndSet(prev, prev.createUnchanged())) {
+            prev = config.get();
+        }
+        return prev.isConfigChanged();
```

Comment:
should probably return the flag for the current generation, not any generation.

Reply:
Fixed